### DISCUSSION TITLE
Welcome to ESP32_New_TimerInterrupt Discussions!

### DISCUSSION THREAD

**Discussion by khoih-prog** [khoih-prog/ESP32_New_TimerInterrupt#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gue21** [comment]
after changing from the older 1.4.1 version to 1.5 or to the "new" 1.2.0 I get following:
_ESP32_New_TimerInterrupt.h:354:82: error: 'TIMER_INTR_T1' was not declared in this scope_
what I am missing? ?

**Comment by khoih-prog** [comment]
You have to read and follow the [**Prerequisites**](https://github.com/khoih-prog/ESP32_New_TimerInterrupt#prerequisites), and install the `ESP32 core v2.0.2`

You are using very old `ESP32 core v1.0.5-`

Good Luck

**Comment by gue21** [comment]
I could not find any instructions to update from v1.0.5 to v2.0.0.
Do I need to de- and reinstall platformIO?

**Reply by khoih-prog** [reply]
Read [Preferences not working with ESP32-S2 and PlatformIO, BUT working with Arduino IDE](https://community.platformio.org/t/preferences-not-working-with-esp32-s2-and-platformio-but-working-with-arduino-ide/25938)

Using PIO, you have to know that ESP32 core v2.0.2 is not supported yet and have to use a workaround 

https://github.com/platformio/platform-espressif32/issues/619#issuecomment-1002306803

If you can't figure out, ask for help in PIO forum, not here.

**Comment by millerp6ATWIT** [comment]
Hello, I've read  much of the documentation but I can't seem to find the purpose of 
`#define _TIMERINTERRUPT_LOGLEVEL_     0`
can you tell me what this does?

**Reply by khoih-prog** [reply]
Check [**Debug**](https://github.com/khoih-prog/ESP32_New_TimerInterrupt#debug) then test by changing the `_TIMERINTERRUPT_LOGLEVEL_` to see the effects.

```
// These define's must be placed at the beginning before #include "ESP32_New_TimerInterrupt.h"
// _TIMERINTERRUPT_LOGLEVEL_ from 0 to 4
// Don't define _TIMERINTERRUPT_LOGLEVEL_ > 0. Only for special ISR debugging only. Can hang the system.
#define TIMER_INTERRUPT_DEBUG         0
#define _TIMERINTERRUPT_LOGLEVEL_     0
```

Next time please create a new Discussion Topic instead of using this [**Welcome to ESP32_New_TimerInterrupt Discussions!**](https://github.com/khoih-prog/ESP32_New_TimerInterrupt/discussions/1)

**Reply by millerp6ATWIT** [reply]
Thank you I will, I just saw one of the examples using 4 so I wasn't sure what it changes.
